components/compliance-service/inspec-agent/resolver/resolver.go

```diff
@@ -339,29 +339,38 @@ func (r *Resolver) handleAwsApiNodesSingleNode(ctx context.Context, m *manager.N
 		awsCreds = managers.GetAWSCreds(secret)
 	}
 
-	nodeInfo := nodeInfo{
+	nodeInfo, tc, secrets := assembleAwsApiNodeInfo(node, m, awsCreds)
+	inspecJob, err := assembleJob(job, nodeInfo, []*inspec.Secrets{&secrets}, tc)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error assembling job. aborting scan for node %s", node.Name)
+	}
+	return []*types.InspecJob{inspecJob}, nil
+}
+
+func assembleAwsApiNodeInfo(node *nodes.Node, m *manager.NodeManager, awsCreds awsec2.AwsCreds) (nodeInfo, inspec.TargetBaseConfig, inspec.Secrets) {
+	nodeDetails := nodeInfo{
 		UUID:           node.Id,
 		CloudID:        m.AccountId,
 		Name:           node.Name,
 		Environment:    "aws-api",
 		ManagerID:      m.Id,
 		CloudAccountID: m.AccountId,
 	}
+	region := awsCreds.Region
```

Comment:
line 359 to 367 here are the real changes.

i wanted to add a unit test to i pulled this bit of node assembling logic out

Reply:
(if no credential id is provided, which is the case for the "read creds from my cloud env" option, awsCreds.Region is blank